modules/search-as-you-type/src/test/java/org/elasticsearch/index/mapper/SearchAsYouTypeFieldMapperTests.java

```diff
@@ -0,0 +1,188 @@
+            assertThat(analyzer.name(), equalTo(analyzerName));
+        }
+    }
+
+    private static void assertEdgeNGramsFieldMapper(Mapper mapper, String analyzerName) {
+        assertThat(mapper, instanceOf(SuggesterizedFieldMapper.class));
+        final SuggesterizedFieldMapper suggesterizedFieldMapper = (SuggesterizedFieldMapper) mapper;
+
+        final SuggesterizedFieldType fieldType = suggesterizedFieldMapper.fieldType();
+        final NamedAnalyzer namedIndexAnalyzer = fieldType.indexAnalyzer();
+        final NamedAnalyzer namedSearchAnalyzer = fieldType.searchAnalyzer();
+        assertThat(namedIndexAnalyzer.analyzer(), instanceOf(SearchAsYouTypeAnalyzer.class));
+        assertThat(namedSearchAnalyzer.analyzer(), not(instanceOf(SearchAsYouTypeAnalyzer.class)));
+        final SearchAsYouTypeAnalyzer indexAnalyzer = (SearchAsYouTypeAnalyzer) fieldType.indexAnalyzer().analyzer();
+
+        for (NamedAnalyzer analyzer : asList(namedIndexAnalyzer, namedSearchAnalyzer)) {
+            assertThat(analyzer.name(), equalTo(analyzerName));
+        }
+
+        assertFalse(indexAnalyzer.isWithShingles());
+        assertTrue(indexAnalyzer.isWithEdgeNGrams());
+
+        assertThat(namedSearchAnalyzer.name(), equalTo(analyzerName));
+    }
+
+    private static void assertShinglesAndEdgeNGramsFieldMapper(Mapper mapper,
+                                                               String analyzerName,
+                                                               int numberOfShingles,
+                                                               boolean withEdgeNGrams) {
+
+        assertThat(mapper, instanceOf(SuggesterizedFieldMapper.class));
+        final SuggesterizedFieldMapper suggesterizedFieldMapper = (SuggesterizedFieldMapper) mapper;
+
+        final SuggesterizedFieldType fieldType = suggesterizedFieldMapper.fieldType();
+        final NamedAnalyzer namedIndexAnalyzer = fieldType.indexAnalyzer();
+        final NamedAnalyzer namedSearchAnalyzer = fieldType.searchAnalyzer();
+        assertThat(namedIndexAnalyzer.analyzer(), instanceOf(SearchAsYouTypeAnalyzer.class));
+        assertThat(namedSearchAnalyzer.analyzer(), instanceOf(SearchAsYouTypeAnalyzer.class));
+        final SearchAsYouTypeAnalyzer indexAnalyzer = (SearchAsYouTypeAnalyzer) fieldType.indexAnalyzer().analyzer();
+        final SearchAsYouTypeAnalyzer searchAnalyzer = (SearchAsYouTypeAnalyzer) fieldType.searchAnalyzer().analyzer();
+
+        for (NamedAnalyzer analyzer : asList(namedIndexAnalyzer, namedSearchAnalyzer)) {
+            assertThat(analyzer.name(), equalTo(analyzerName));
+        }
+
+        for (SearchAsYouTypeAnalyzer analyzer : asList(indexAnalyzer, searchAnalyzer)) {
+            assertTrue(analyzer.isWithShingles());
+            assertThat(analyzer.getShingleSize(), equalTo(numberOfShingles));
+        }
+
+        assertThat(indexAnalyzer.isWithEdgeNGrams(), equalTo(withEdgeNGrams));
+    }
```

Comment:
Can you add some tests to check the parsing and the output of the documents (make sure that all shingles/edge ngrams are correctly created) ?

Reply:
++